app/vmalert/main.go

```diff
@@ -361,7 +361,6 @@ func configReload(ctx context.Context, m *manager, groupsCfg []config.Group, sig
 			// set success to 1 since previous reload
 			// could have been unsuccessful
 			configSuccess.Set(1)
```

Comment:
Base on the names, should we update `configTimestamp` as well? 
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/fc01167962c5ee4fe0052d3a47fc18e29b24f865/app/vmalert/main.go#L204-L205
And I'm little confused about `vmalert_config_last_reload_successful`'s meaning and the comment under `configsEqual`
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/fc01167962c5ee4fe0052d3a47fc18e29b24f865/app/vmalert/main.go#L361-L362
We set `configSuccess` to 0  when last newGroupsCfg update failed for some wrong configuration[like contains recording rules but `-remoteWrite.url` isn't set]. But in next checkInterval, `configSuccess` will be set to 1 if config doesn't change, although it's still wrong and not be applied and configErr is not nil. 
cc @hagen1778

Reply:
I'm not sure I got it. 
the `newGroupsCfg` is loaded from the configured destination on each iteration. If destination returns bad config -  `newGroupsCfg` will be bad on each iteration.
The `groupsCfg` represents current valid config. It gets updated only if `newGroupsCfg` was valid as well.
So if `newGroupsCfg` was bad, then `groupsCfg != newGroupsCfg` and metric isn't supposed to get updated.

Reply:
Yeah, fingured that when reviewing https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4790)